cupy/cuda/__init__.py

```diff
@@ -14,48 +14,53 @@
 from cupy_backends.cuda.api import runtime  # NOQA
 from cupy_backends.cuda.libs import cublas  # NOQA
 from cupy_backends.cuda.libs import curand  # NOQA
+from cupy_backends.cuda.libs import cusolver  # NOQA
 from cupy_backends.cuda.libs import cusparse  # NOQA
 from cupy_backends.cuda.libs import nvrtc  # NOQA
 from cupy_backends.cuda.libs import profiler  # NOQA
 
 
 _available = None
 
+
+class UnavailableModule():
+    available = False
+
+# All `*_enabled` flags are kept just for backward compatibility.
```

Comment:
👍 for keeping the old flags around. I wonder if it's possible to emit a warning when users attempt to access the old flags? Like "the xxxxx_enabled flag is deprecated and will be removed in a future release. Please use xxxxx.available instead."

Reply:
This question for `CUB_DISABLED` is related: https://github.com/cupy/cupy/pull/3562#issuecomment-657793261

Reply:
As for `CUB_DISABLED`, I think we don't have to emit depreaction as it was undocumented.

Reply:
Added DeprecationWarnings.